chore: extract RPCs to docs (#228)
nervosnetwork/godwoken-web3: docs/apis.md

- eth_sendTransaction (only wallet client can do this)
- eth_sendRawTransaction (for compatible reason, we use `gw_submit_l2Transaction` and `poly_submitL2Transaction` for submitting transactions)

## Additional Modules

### gw (Godwoken RPCs)

- gw_ping
- gw_get_tip_block_hash
- gw_get_block_hash
- gw_get_block
- gw_get_block_by_number
- gw_get_balance
- gw_get_storage_at
- gw_get_account_id_by_script_hash
- gw_get_nonce
- gw_get_script
- gw_get_script_hash
- gw_get_data
- gw_get_transaction_receipt
- gw_execute_l2transaction
- gw_execute_raw_l2transaction
- gw_submit_l2transaction
- gw_submit_withdrawal_request

### Usage

checkout [Godwoken Docs](https://github.com/nervosnetwork/godwoken/blob/develop/docs/RPC.md)

### poly (Polyjuice RPCs)

- poly_getEthAddressByGodwokenShortAddress
- poly_getChainInfo
- poly_getDefaultFromAddress
- poly_getContractValidatorTypeHash
- poly_getRollupTypeHash
- poly_getRollupConfigHash
- poly_getEthAccountLockHash
- poly_getCreatorId
- poly_submitL2Transaction
- poly_executeRawL2Transaction

### Usage

#### poly_executeRawL2Transaction

> This method is similar to the concept of `"eth_call"` in Ethereum.

> all most the same with `gw_execute_raw_l2Transaction`, but the first param `serialized raw l2Transaction` use different molecule structure with an additional `addressMapping` filed.

Reviewer: ```suggestion
`poly_executeRawL2Transaction` and `gw_execute_raw_l2Transaction` are almost the same, while the first param `serializedRawL2Transaction` of `poly_executeRawL2Transaction` uses different molecule structure with an additional `addressMapping` filed.
```

- gw_execute_raw_l2Transaction
  - params: [serializedRawL2Transaction, blockNumber]
  - how to serialize: [code](https://github.com/nervosnetwork/polyjuice-provider/blob/main/packages/base/src/util.ts#L465-L468)
- poly_executeRawL2Transaction
  - params: [serializedRawL2TransactionWithAddressMapping, blockNumber]
  - how ot serialize: [code](https://github.com/nervosnetwork/polyjuice-provider/blob/main/packages/base/src/util.ts#L326-L335)

#### poly_submitL2Transaction

> This method is similar to the concept of `"eth_sendRawTransaction"` in Ethereum.

> all most the same with `gw_submit_l2Transaction`, but the first param `serialized l2Transaction` use different molecule structure with an additional `addressMapping` filed.

Reviewer: ```suggestion
`poly_submitL2Transaction` is almost the same with `gw_submit_l2Transaction`, but its first param `serializedL2TransactionWithAddressMapping` use different molecule structure with an additional `addressMapping` filed.
```
